ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlClient.java

```diff
@@ -44,27 +45,50 @@
   private final HttpClient httpTlsClient;
   private final LocalProperties localProperties;
   private final Optional<String> basicAuthHeader;
+  private final Optional<HostAliasResolver> hostAliasResolver;
 
   public KsqlClient(
       final Map<String, String> clientProps,
       final Optional<BasicCredentials> credentials,
       final LocalProperties localProperties,
-      final HttpClientOptions httpClientOptions
+      final HttpClientOptions httpClientOptions,
+      final Optional<HostAliasResolver> hostAliasResolver
```

Comment:
It seems a shame to require every use of this _production_ class to supply a `HostAliasResolver`, given this should only be passed for _test_ cases, right?  And most test cases won't need this either. Just one or two specific tests that are testing the new code you're adding. Yet `HostAliasResolver` is mentioned around 40 times in this PR.

Is there another way?  

At the very least you could remove this parameter from this constructor and add another constructor, annotated with `@VisibleForTesting` that takes the resolver.  However, ideally, the production code should know nothing of `HostAliasResolver`...

I think this can be achieved by injecting an interface to handle the `SocketAddress.inetSocketAddress` call.   So rather than taking a `HostAliasResolver` the `@VisiableForTesting` constructor takes `BiFunction<Integer, String, SocketAddress> socketAddressFactory`.  Test code can then use this to perform the same aliasing that `HostAliasResolver` does now.   Please see if you can remove `HostAliasResolver` from the production code base, or at least from the non- `@VisibleForTesting` constructors.




SocketAddress.inetSocketAddress(server.getPort(), aliasHost)